File: 3264.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> nums(n);
    for (int i = 0; i < n; i++) {
        cin >> nums[i];
    }

    int k, multiplier;
    cin >> k >> multiplier;

    for (int i = 0; i<k; i++) {
        int min = nums[0];
        int index = 0;
        for (int j = 0; j < n; j++) {
            if (nums[j] < min) {
                min = nums[j];
                index = j;
            }
        }
            
        nums[index]*=multiplier;
    }

    for (int x : nums) {
        cout << x << " ";
    }
}

File: 3264.py
nums = list(map(int, input().split()))
k = int(input())
multiplier = int(input())

for i in range(k):
    mn = min(nums)
    for j in range(len(nums)):
        if nums[j] == mn:
            nums[j] *= multiplier
            break

print(*nums)